zulipterminal/config/gruvbox.py

```diff
@@ -0,0 +1,73 @@
+from enum import Enum
+
+from zulipterminal.config.themes import add_color_prop
+
+
+# fmt: off
+class Color(Enum):
+	 BLACK         = 'h234   #1d2021'
```

Comment:
Though the current version makes it easier for theme contributor to write their colors. Having to specify something like ZTCOLOR('h234', '#1d2021') is a bit more verbose.

Reply:
For simple tuples or even a namedtuple we wouldn't need that prefix.

I agree we need the parentheses and quotes with a tuple approach, but that doesn't seem overly demanding?

Reply:
I think using a string is the cleanest looking of all options. Using a tuple approach would make it use many quotes now that even 16 color name would be included.
Also, is the suggestion to replace enums with tuples/namedtuples? If so it would not be possible to import them for use in other themes. It has to either be a dict or a class. Using a dict would make the STYLE entries less cleaner than the current method.

Reply:
For a tuple it would be like
```python3
class Color(Enum):
    BLACK = ("black", "h234", "#1d2021")
```
If a namedtuple was involved, we'd be able to have `Color.BLACK.in_16_color` or similar.

Reply:
| Tuple | String |
| :-----: | :----: |
| <img src='https://user-images.githubusercontent.com/64144419/121656121-c7390f80-cabc-11eb-87d6-456ec115a7ba.png' height=200> | <img src='https://user-images.githubusercontent.com/64144419/121656213-e59f0b00-cabc-11eb-9d09-8d901302beba.png' height=200> |

Both are fine, depends on ones liking IG.
(I'm leaning towards single string maybe because less quotes reduces clutter?)

Reply:
> If a namedtuple was involved, we'd be able to have Color.BLACK.in_16_color or similar.

With string, it can always be parsed into a namedtuple by the factory, no problem. The theme config is the user-facing part, where they might want to modify. While it is just small detail, the less cluttered single string definitely reduces the effort for theme contributors, makes it less intimidating.

I do think the tuple version is equally fine. It's just that namedtuple is my least preferred option for the user-facing file.

Reply:
Technically the tuple and string alternatives are not being compared like with like, since urwid colors can have spaces - they don't have underscores except where you've added them here.

I agree the text is less cluttered, and we can always process it. I believe we can eventually make this processing occur in the enum/tuple though.

That said, I'm somewhat wary of going too far into catering for customization since I'd like to just get this iterating and tested as being equivalent to current themes. Once themes are reformatted as multi-depth colors with simple styles and a urwid-palette factory then we can iterate on the representation with less concern.

Note that we can use namedtuple without it needing to be explicit in each enum entry.

Reply:
Yeah, tuples are more maintainable. It is relatively hard to maintain the spacing and align the strings.

Reply:
The spacing issue is a one time thing right? After that any new change would require only handling one line. Also, the alignment issue is same for tuples also right? Unaligned-tuples wouldn't be very readable.
I'm fine with tuples also though :)

Reply:
I'll mark this as resolved? If Tuples feel more reasonable at a later point we could shift. Also to be considered is using Tuples would mean we have to suppress another flake8 warnings related to "too much spacing in a tuple" I think.